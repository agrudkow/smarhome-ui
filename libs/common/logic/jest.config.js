module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
