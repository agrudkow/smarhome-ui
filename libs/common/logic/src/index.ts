export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
