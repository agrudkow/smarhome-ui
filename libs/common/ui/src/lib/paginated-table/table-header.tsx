import React, { PropsWithChildren, CSSProperties } from 'react';
import {
  makeStyles,
  createStyles,
  TableHead,
  TableRow,
  TableCell,
  TableSortLabel,
  Size,
} from '@material-ui/core';

const useStyles = makeStyles(() =>
  createStyles({
    visuallyHidden: {
      border: 0,
      clip: 'rect(0 0 0 0)',
      height: 1,
      margin: -1,
      overflow: 'hidden',
      padding: 0,
      position: 'absolute',
      top: 20,
      width: 1,
    },
  })
);

export enum Order {
  asc = 'asc',
  desc = 'desc',
}

export interface HeadCell<Keys extends string | symbol | number> {
  disablePadding: boolean;
  id: Keys;
  label: string;
  numeric: boolean;
  enableSorting?: boolean;
  style?: CSSProperties;
}

export interface TableHeaderProps<Keys extends string | symbol | number> {
  onRequestSort: (property: Keys) => void;
  order: Order;
  orderBy: Keys;
  headCells: HeadCell<Keys>[];
  rowsAlignment: Record<Keys, 'left' | 'right'>;
}

export type TableHeaderFunctionComponentType = <
  Keys extends string | symbol | number
>(
  props: PropsWithChildren<TableHeaderProps<Keys>>
) => JSX.Element;

export const TableHeader: TableHeaderFunctionComponentType = ({
  order,
  orderBy,
  headCells,
  rowsAlignment,
  onRequestSort,
}) => {
  const classes = useStyles();

  const createSortHandler = (
    property: Parameters<typeof onRequestSort>[0]
  ) => () => {
    onRequestSort(property);
  };

  return (
    <TableHead>
      <TableRow>
        {headCells.map((headCell, index) => (
          <TableCell
            key={`table-cell-${headCell.id}-${index}`}
            align={rowsAlignment[headCell.id]}
            padding={headCell.disablePadding ? 'none' : 'default'}
            sortDirection={orderBy === headCell.id ? order : false}
            style={headCell.style}
          >
            {headCell.enableSorting ? (
              <TableSortLabel
                active={orderBy === headCell.id}
                direction={orderBy === headCell.id ? order : Order.asc}
                onClick={createSortHandler(headCell.id)}
              >
                {headCell.label}
                {orderBy === headCell.id ? (
                  <span className={classes.visuallyHidden}>
                    {order === Order.desc
                      ? 'sorted descending'
                      : 'sorted ascending'}
                  </span>
                ) : null}
              </TableSortLabel>
            ) : (
              headCell.label
            )}
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
};

export default TableHeader;
