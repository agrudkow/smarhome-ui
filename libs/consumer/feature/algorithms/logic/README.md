# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
