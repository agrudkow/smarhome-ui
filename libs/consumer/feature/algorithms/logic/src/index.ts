export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
