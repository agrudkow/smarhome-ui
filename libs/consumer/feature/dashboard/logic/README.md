# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
