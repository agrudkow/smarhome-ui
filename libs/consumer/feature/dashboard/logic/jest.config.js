module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
