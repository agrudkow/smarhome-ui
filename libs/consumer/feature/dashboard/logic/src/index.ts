export * from './lib/consumer-feature-dashboard-logic';
