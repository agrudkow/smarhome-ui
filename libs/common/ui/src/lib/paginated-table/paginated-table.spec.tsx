import React from 'react';
import { customRender } from '@smarthome/common/logic';
import PaginatedTable from './paginated-table';
import { Cell } from './table-header';

const rows = [
  {
    id: 'test-1',
    name: 'test-name',
  },
  {
    id: 'test-2',
    name: 'test-name',
  },
  {
    id: 'test-3',
    name: 'test-name',
  },
];

const cells: Cell<'id' | 'name'>[] = [
  {
    disablePadding: false,
    id: 'id',
    align: 'center',
    label: 'Test',
  },
  {
    disablePadding: false,
    id: 'name',
    align: 'center',
    label: 'Test',
  },
];

describe(' PaginatedTable', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <PaginatedTable<typeof rows[0]> data={rows} cells={cells} />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot (without footer)', () => {
    const { baseElement } = customRender(
      <PaginatedTable<typeof rows[0]>
        data={rows}
        cells={cells}
        footer={false}
      />
    );
    expect(baseElement.firstChild).toMatchSnapshot();
  });
});
