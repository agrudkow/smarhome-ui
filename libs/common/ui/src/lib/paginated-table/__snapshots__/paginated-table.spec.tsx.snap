// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` PaginatedTable should match snapshot (without footer) 1`] = `
.c0 .MuiPaper-root {
  margin: 0;
  padding: 5px 0;
  background-color: rgba(255,255,255,1);
}

.c0 .MuiPaper-rounded {
  border-radius: 6px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="MuiPaper-root makeStyles-paper-4 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="table"
        class="MuiTable-root makeStyles-table-5"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              <b>
                Test
              </b>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              <b>
                Test
              </b>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              test-1
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              test-name
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              test-2
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              test-name
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              test-3
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              test-name
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
            style="height: 406px;"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
              colspan="2"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
